contracts/OvertimeVaults/BaseSportVault.sol

```diff
@@ -0,0 +1,335 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts-upgradeable/token/ERC20/utils/SafeERC20Upgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import "@openzeppelin/contracts-upgradeable/security/PausableUpgradeable.sol";
+
+import "../utils/proxy/solidity-0.8.0/ProxyReentrancyGuard.sol";
+import "../utils/proxy/solidity-0.8.0/ProxyOwned.sol";
+
+import "../interfaces/ISportsAMM.sol";
+import "../interfaces/ISportPositionalMarket.sol";
+
+contract BaseSportVault is Initializable, ProxyOwned, PausableUpgradeable, ProxyReentrancyGuard {
+    /* ========== LIBRARIES ========== */
+    using SafeERC20Upgradeable for IERC20Upgradeable;
+
+    struct DepositReceipt {
+        uint round;
+        uint amount;
+    }
+
+    /* ========== CONSTANTS ========== */
+    uint private constant HUNDRED = 1e20;
+    uint private constant ONE = 1e18;
+
+    /* ========== STATE VARIABLES ========== */
+
+    ISportsAMM public sportsAMM;
+    IERC20Upgradeable public sUSD;
+
+    bool public vaultStarted;
+
+    uint public round;
+    uint public roundLength;
+    mapping(uint => uint) public roundStartTime;
+
+    mapping(uint => address[]) public usersPerRound;
+    mapping(uint => mapping(address => bool)) public userInRound;
+
+    mapping(uint => mapping(address => uint)) public balancesPerRound;
+    mapping(address => bool) public withdrawalRequested;
+    mapping(address => DepositReceipt) public depositReceipts;
+
+    mapping(uint => uint) public allocationPerRound;
+
+    mapping(uint => address[]) public tradingMarketsPerRound;
+    mapping(uint => mapping(address => ISportsAMM.Position)) public tradingMarketPositionPerRound;
+    mapping(uint => mapping(address => bool)) public isTradingMarketInARound;
+
+    mapping(uint => uint) public profitAndLossPerRound;
+
+    uint public maxAllowedDeposit;
+    uint public utilizationRate;
+
+    mapping(uint => uint) public capPerRound;
+
+    uint public minDepositAmount;
+
+    uint public maxAllowedUsers;
+    uint public usersCurrentlyInVault;
+
+    //TODO: add staking thales so gamified staking bonuses are updated on every round closing
```

Comment:
this is not implemented?

Reply:
no, it needs more thought, but I dont want to forget about it :)